big-o-notation/big-o.js

```diff
@@ -0,0 +1,47 @@
+/* exported uniqueQuadratic, uniqueLinear */
+
+function uniqueLinear(words) {
+  const seen = {}; // 1 * 1 = O(1)
+  const unique = []; // 1 * 1 = O(1)
+  for (
+    let i = 0; // 1 * 1 = O(1)
+    i < words.length; // 2 * n = O(2n)
+    i++ // 2 * n = O(2n)
+  ) {
+    const word = words[i]; // 1 * 1 = O(1)
+    if (!seen[word]) {
+      // 3 * n = O(3n)
+      seen[word] = true;
+      unique[unique.length] = word; // 2 * n = O(2n)
+    }
+  }
+  return unique;
+} // Big O Notation for uniqueLinear: O(n)
+
+function uniqueQuadratic(words) {
+  const unique = []; // 1 * 1 = O(1)
+  for (
+    let i = 0; // 1 * 1 = O(1)
+    i < words.length; // 2 * n = O(2n)
+    i++ // 2 * n = O(2n)
+  ) {
+    const word = words[i]; // 1 * 1 = O(1)
+    let isUnique = true; // 1 * 1 = O(1)
+    for (
+      let c = 0; // 1 * 1 = O(1)
+      c < i; // 2 * n = O(2n)
+      c++ // 2 * n = O(2n)
+    ) {
+      const comparing = words[c]; // 1 * 1 = O(1)
+      if (comparing === word) {
+        // 2 * n = O(2n)
+        isUnique = false; // 1 * 1 = O(1)
+      }
+    }
+    if (isUnique) {
+      // 1 * 1 = O(1)
```

Comment:
This is still inside the outer loop so it will run once for every item in the data set, so it will be `O(n)`